src/postfork.cpp

```diff
@@ -136,6 +116,13 @@ bool set_child_group(job_t *j, pid_t child_pid) {
         if (j->pgid == -2) {
             j->pgid = child_pid;
         }
+        // The parent sets the child's group. This incurs the well-known unavoidable race with the
```

Comment:
Reminder to either of us: clean up the comments above the `set_child_group` function to reflect the new behavior. (GitHub won't let me comment there)